### PR TITLE
Add missed parameters for the Azure Cosmos DB Account

### PR DESCRIPTION
**Pull request by mdanylyuk**
# [1670](https://github.com/aztfmod/terraform-azurerm-caf/issues/1670)

## PR Checklist

---

- [x] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Does this introduce a breaking change

- [ ] YES
- [x] NO

I added new parameters for the managing security in the Cosmos DB Account (https://azure.github.io/cloud-scale-data-for-devs-guide/security.html#role-based-access-control)

### CODE REVIEW

File: modules/databases/cosmos_dbs/cosmosdb_account.tf [hunk 1]
```diff
@@ -17,13 +17,15 @@ resource "azurerm_cosmosdb_account" "cosmos_account" {
...
-  create_mode                       = try(var.settings.create_mode, null)
-  public_network_access_enabled     = try(var.settings.public_network_access_enabled, true)
+  enable_free_tier                    = try(var.settings.enable_free_tier, false)
+  ip_range_filter                     = try(var.settings.ip_range_filter, null)
+  enable_multiple_write_locations     = try(var.settings.enable_multiple_write_locations, false)
+  enable_automatic_failover           = try(var.settings.enable_automatic_failover, null)
+  is_virtual_network_filter_enabled   = try(var.settings.is_virtual_network_filter_enabled, null)
+  create_mode                         = try(var.settings.create_mode, null)
+  public_network_access_enabled       = try(var.settings.public_network_access_enabled, true)
+  access_key_metadata_writes_enabled  = try(var.settings.access_key_metadata_writes_enabled, true)
```
Comment (arnaudlh) [inline modules/databases/cosmos_dbs/cosmosdb_account.tf:27]:
```suggestion
  access_key_metadata_writes_enabled  = try(var.settings.access_key_metadata_writes_enabled, null)
```

File: modules/databases/cosmos_dbs/cosmosdb_account.tf [hunk 2]
```diff
@@ -17,13 +17,15 @@ resource "azurerm_cosmosdb_account" "cosmos_account" {
...
+  local_authentication_disabled       = try(var.settings.local_authentication_disabled, false)
```
Comment (arnaudlh) [inline modules/databases/cosmos_dbs/cosmosdb_account.tf:28]:
```suggestion
  local_authentication_disabled       = try(var.settings.local_authentication_disabled, null)
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
thanks for the PR @mdanylyuk.  I'd suggest we put ```null``` instead of a module enforced default value. We learnt that we want to let the provider default and usually not enforce an overide at module level.

File: modules/databases/cosmos_dbs/cosmosdb_account.tf (hunk 1 above)
Comment (mdanylyuk) [reply modules/databases/cosmos_dbs/cosmosdb_account.tf]: Ok, changed it. But what's interesting there are lots other parameters with "true/false" as default value)

File: modules/databases/cosmos_dbs/cosmosdb_account.tf (hunk 2 above)
Comment (mdanylyuk) [reply modules/databases/cosmos_dbs/cosmosdb_account.tf]: Changed

**Review by arnaudlh — APPROVED** [review]
LGTM

File: modules/databases/cosmos_dbs/cosmosdb_account.tf (hunk 1 above)
Comment (arnaudlh) [reply modules/databases/cosmos_dbs/cosmosdb_account.tf]: mistakes from the past ;)

**Comment by arnaudlh** [comment]
thanks for your contribution @mdanylyuk

**Pull request merged**